### PR TITLE
bugfix #2245 shipping step shipping methods issues

### PR DESCRIPTION
**Pull request by tatiana-scandi**
Fixes 4 issues mentioned in the [bug](https://github.com/scandipwa/scandipwa/issues/2245).

### ISSUE 1:

Problem: Shipping methods are not loaded when user goes to billing step and than goes back.
Solution: Add `shippingMethod` field to `Checkout` reducer, get data from it when user goes back.

### ISSUE 2:
Problem: Upon going to the checkout page, for some reason it requests the US shipping methods three times.
Solution: Move `this.estimateShipping()` to `componentDidMount` from `_construct` => now only one API call on page load. 

### ISSUE 3:
Problem: `Region_id` is not updated when changing country.
Solution: Add`setState` for `regionId`  on country update.

### ISSUE 4:
Problem: Incomplete data in API request fetching shipping methods available
Actual problem: `setState` inside `debounce` causes state to be updated for one field only if browser autofill is used. Actual field values are missing in the state and therefore in the API request. Yet not using `debounce` at all is not a good solution, as it leads API call to be sent with every letter typed.
Solution: "Debounce" (i.e. wait a bit before calling the method and call it only once) `estimateShipping`, not `setState`.

### CODE REVIEW

**Comment by aleksandrakorolova** [comment]
**Steps:**
1. Open product page
2. Add to cart
3. Open shipping step
4. Select Canada 
5. Fill form
6. Go to billing step
7. Return to Shipping step

**Actual result:**
Two graphql are received:
1. With default country and State 


2. With selected country and State  + zip code, city ... are missing



**Expected result**
One graphql is received with selected Country (CA) and state, also zip code, city and etc

**Comment by tatiana-scandi** [comment]
@aleksandrakorolova fixed, please recheck. 2 cases have to be checked separately: 1. region as dropdown (e.g. US) 2. region as input (e.g. Belarus)

**Comment by aleksandrakorolova** [comment]
Now we received two equal requests if user going back from billing to shipping step (the first time). If he repeats it, one request is received

File: packages/scandipwa/src/component/MyAccountAddressForm/MyAccountAddressForm.component.js [hunk 1]
```diff
@@ -104,14 +104,22 @@ export class MyAccountAddressForm extends FieldForm {
...
-            isStateRequired: is_state_required,
-            availableRegions: available_regions || []
+            availableRegions: available_regions || [],
+            isStateRequired: is_state_required
         });
+
+        // avoid region reset when coming back to shipping step
+        if (prevCountryId && prevCountryId !== countryId) {
+            this.setState({
+                regionId: available_regions ? available_regions[0].id : null
```
Comment (IvansZuks) [inline packages/scandipwa/src/component/MyAccountAddressForm/MyAccountAddressForm.component.js:120]: better to destruct it

File: packages/scandipwa/src/component/CheckoutAddressForm/CheckoutAddressForm.component.js [hunk 2]
```diff
@@ -30,28 +30,40 @@ export class CheckoutAddressForm extends MyAccountAddressForm {
...
         const {
-            address: { region: { region = '' } = {} }
+            shippingFields: {
+                city = '',
+                region_id: regionId = null,
+                region_string: region = '',
+                country_id: countryId = '',
+                postcode = ''
+            }
         } = this.props;
```
Comment (IvansZuks) [inline packages/scandipwa/src/component/CheckoutAddressForm/CheckoutAddressForm.component.js:46]: I'm not sure but github shows all next lines with some strange color, like it's all in the string. Please make double check that this code is okay

**Review by IvansZuks — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/component/CheckoutAddressForm/CheckoutAddressForm.component.js (hunk 2 above)
Comment (tatiana-scandi) [reply packages/scandipwa/src/component/CheckoutAddressForm/CheckoutAddressForm.component.js:46]: Checked, it's fine.

File: packages/scandipwa/src/component/MyAccountAddressForm/MyAccountAddressForm.component.js (hunk 1 above)
Comment (tatiana-scandi) [reply packages/scandipwa/src/component/MyAccountAddressForm/MyAccountAddressForm.component.js]: Added default value of `[]` for `available_regions`.

```regionId: available_regions[0]?.id || null```

Can I destruct it more? Maybe I didn't get the idea completely. Using array destructing looks like overkill here.

File: packages/scandipwa/src/component/MyAccountAddressForm/MyAccountAddressForm.component.js [hunk 3]
```diff
@@ -104,14 +104,22 @@ export class MyAccountAddressForm extends FieldForm {
...
-            isStateRequired: is_state_required,
-            availableRegions: available_regions || []
+            availableRegions: available_regions || [],
+            isStateRequired: is_state_required
         });
+
+        // avoid region reset when coming back to shipping step
+        if (prevCountryId && prevCountryId !== countryId) {
+            this.setState({
+                regionId: available_regions[0]?.id || null
```
Comment (IvansZuks) [inline packages/scandipwa/src/component/MyAccountAddressForm/MyAccountAddressForm.component.js:120]: Please desctruct available_regions

**Review by IvansZuks — APPROVED** [review]
Fix 1 comment and can merge

**Pull request merged**